test/CP/variables/IntVar.jl

```diff
@@ -39,4 +39,18 @@
 
         @test SeaPearl.assignedValue(x) == 5
     end
+
+    @testset "-()" begin
+        trailer = SeaPearl.Trailer()
+        x = SeaPearl.IntVar(5, 8, "x", trailer)
+        y = -x
+        
+        @test isa(y, SeaPearl.IntDomainViewOpposite)
+        @test SeaPearl.minimum(y) == -8
+        @test SeaPearl.maximum(y) == -5
+
+        SeaPearl.assign!(x, 5)
+
+        @test SeaPearl.length(y) == 1

```

Comment:
Nice tests. I'd just add a test on the value of `y` after assigning x.

Reply:
Actually, y is a view on the domain, so I don't think there is a way to test the value, but maybe it would be more logical to return a  `IntVarViewOpposite`, in which case we could test the value.

Reply:
We could just test the minimum and maximum actually.

Reply:
I thought you were returning ` IntVarViewOpposite` in fact. I'm wondering what is the more logical way to think about it, at first sight, I'd say that `-` should generate a `IntVarViewOpposite`, we should consider a little practical case.

Reply:
Recall that the goal of overloading the `-` operator is to facilitate the model declaration, this feature should generate an object that can be handled by the user. `IntDomainViewOpposite` is just a technical object relative to the operation of the `IntVarViewOpposite` variable.
 
```julia
    trailer = SeaPearl.Trailer()
    model = SeaPearl.CPModel(trailer)
    x = SeaPearl.IntVar(2, 3, "x", trailer)
    y = - x                 #y = SeaPearl.IntVarViewOpposite(x, "-x")
    SeaPearl.addVariable!(model, x)
    SeaPearl.addVariable!(model, y)
    ...
    SeaPearl.addConstraint!(model, SeaPearl.SumToZero([x, y], trailer))
```